port/unix/omrsysinfo.c

```diff
@@ -7354,3 +7376,82 @@ get_Dispatch_IstreamCount(void) {
+		if (stat(procDir, &st) == OMRPORT_SYSINFO_STAT_FAILURE) {
+			Trc_PRT_sysinfo_get_process_start_time_stat_error((unsigned long long)pid);
+			goto done;
+		}
+		processStartTimeInNanoseconds = (uint64_t)st.st_mtime * OMRPORT_TIME_DELTA_IN_NANOSECONDS + st.st_mtim.tv_nsec;
+#elif defined(OSX) /* defined(LINUX) */
+		int mib[OMRPORT_SYSINFO_NUM_SYSCTL_ARGS] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, pid};
+		size_t len = sizeof(struct kinfo_proc);
+		struct kinfo_proc procInfo;
+		if (sysctl(mib, OMRPORT_SYSINFO_NUM_SYSCTL_ARGS, &procInfo, &len, NULL, 0) == OMRPORT_SYSINFO_SYSCTL_FAILURE) {
+			Trc_PRT_sysinfo_get_process_start_time_sysctl_error((unsigned long long)pid);
+			goto done;
+		}
+		if (0 == len) {
+			Trc_PRT_sysinfo_get_process_start_time_pid_does_not_exist((unsigned long long)pid);
+			goto done;
+		}
+		processStartTimeInNanoseconds =
+			((uint64_t)procInfo.kp_proc.p_starttime.tv_sec * OMRPORT_TIME_DELTA_IN_NANOSECONDS) +
+			((uint64_t)procInfo.kp_proc.p_starttime.tv_usec * OMRPORT_SYSINFO_NANOSECONDS_PER_MICROSECOND);
+#elif defined(AIXPPC) /* defined(OSX) */
+		pid_t convertedPid = (pid_t)pid;
+		struct procsinfo procInfos[] = {0};
+		int ret = getprocs(procInfos, sizeof(procInfos[0]), NULL, 0, &convertedPid, sizeof(procInfos) / sizeof(procInfos[0]));
+		if (-1 == ret) {
+			Trc_PRT_sysinfo_get_process_start_time_getprocs_error((unsigned long long)pid);
+			goto done;
+		} else if (0 == ret) {
+			Trc_PRT_sysinfo_get_process_start_time_pid_does_not_exist((unsigned long long)pid);
+			goto done;
+		}
+		processStartTimeInNanoseconds = (uint64_t)(procInfos[0].pi_start) * OMRPORT_TIME_DELTA_IN_NANOSECONDS;
+#elif defined(J9ZOS390) /* defined(AIXPPC) */
+		pgtha pgtha;
+		ProcessData processData;
+		pgthc *currentProcessInfo = NULL;
+		uint32_t dataOffset = 0;
+		uint32_t inputSize = sizeof(pgtha);
+		unsigned char *input = (unsigned char *)&pgtha;
+		uint32_t outputSize = sizeof(ProcessData);
+		unsigned char *output = (unsigned char *)&processData;
+		uint32_t ret = 0;
+		uint32_t retCode = 0;
+		uint32_t reasonCode = 0;
+		memset(input, 0, sizeof(pgtha));
+		memset(output, 0, sizeof(processData));
+		pgtha.pid = pid;
+		pgtha.accesspid = PGTHA_ACCESS_CURRENT;
+		pgtha.flag1 = PGTHA_FLAG_PROCESS_DATA;
+		BPXNGTH(&inputSize, &input, &outputSize, &output, &ret, &retCode, &reasonCode);
+		if (-1 == ret) {
+#if defined(_LP64)
+			Trc_PRT_sysinfo_get_process_start_time_BPX4GTH_error((unsigned long long)pid);
+#else /* defined(_LP64) */
+			Trc_PRT_sysinfo_get_process_start_time_BPX1GTH_error((unsigned long long)pid);
+#endif /* defined(_LP64) */
+			goto done;
+		}
+		dataOffset = *((unsigned int *)processData.pgthb.offc);
+		dataOffset = (dataOffset & I_32_MAX) >> 8;
+		currentProcessInfo = (pgthc *)(((char *)&processData) + dataOffset);
+		processStartTimeInNanoseconds = (uint64_t)currentProcessInfo->starttime * OMRPORT_TIME_DELTA_IN_NANOSECONDS;
+#endif /* defined(LINUX) */
```

Comment:
Add an `#else` block, and print a message such as `#warning omrsysinfo_get_process_start_time not implemented on platform`. Once this PR is merged, we will need to open an issue requesting the z/TPF team to add an implementation for the z/TPF platform.

Reply:
I opened an issue to request the z/TPF team to add an implementation of `omrsysinfo_get_process_start_time` for the z/TPF platform https://github.com/eclipse/omr/issues/7226.